internal/action/command.go

```diff
@@ -589,7 +591,9 @@ func SetGlobalOptionNative(option string, nativeValue interface{}) error {
 	}
 
 	for _, b := range buffer.OpenBuffers {
-		b.SetOptionNative(option, nativeValue)
+		if err := b.SetOptionNative(option, nativeValue); err != nil {
+			return err
+		}
```

Comment:
Hmm... this unexpectedly overrides local per-buffer settings that were changed at runtime by the user via `setlocal` or by a plugin?

Reply:
Uff, that was a nasty one...as a result of `filetype` changes.

Reply:
I'm not sure why are you mentioning `filetype` here...

Anyway... Let's first focus on the most basic cases. What should be the behavior of `set` after `setlocal`? Should `set` set the option value for all buffers, even those whose local option value was changed by `setlocal` or by a plugin? It seems that yes, it should just set it for all buffers. It is both simpler and more intuitive.

And in this regard, I see that the latest version of your PR is buggy: `set` doesn't change local settings if the value is sets is the same as the current global setting. E.g. if `tabsize` is currently set to 4:

1. `setlocal tabsize 8`
2. `set tabsize 4` -> does NOT change the current buffer's setting from 8 back to 4

whereas the following is working as expected:

1. `setlocal tabsize 8`
2. `set tabsize 12` -> changes the current buffer's setting from 8 to 12

...Now, do we want `reload` and `set filetype` to behave the same way as `set` in this regard? Probably no, we want them to respect locally overridden settings. Now, answering your question: neither of the 2 behaviors you mentioned seems the right one to me. Both seem like unreliable heuristics. If the user or a plugin sets a buffer's local option to the same value as the global option, it doesn't mean it actually wants to reset it to the global value, it may be just a coincidence.

It seems we should rather explicitly track whether the given option was overridden locally for the given buffer at least once since the last time it was set globally.